pkg/image/layer.go

```diff
@@ -80,7 +81,8 @@ func (l *Layer) uncompressedTarCache(uncompressedLayersCacheDir string) (string,
 // file tree, and the layer squash tree.
 func (l *Layer) Read(catalog *FileCatalog, imgMetadata Metadata, idx int, uncompressedLayersCacheDir string) error {
 	var err error
-	l.Tree = filetree.NewFileTree()
+	tree := filetree.New()
```

Comment:
Is there any difference here doing `l.Tree = filetree.New()` and then passing `l.tree` to downstream functions?

I noticed `NewSearchContext()` takes `l.Tree`but things like `squashfsVisitor takes tree`

Reply:
indeed, this is an important change. `Layer.tree` has been changed from `*filetree.FileTree` into a `filetree.Reader`, so no mutations are allowed. This change allows us to have a local instance that is mutable for indexing during the Read() call, but only persist and object that is read-only.